docs/book/src/tasks/experimental-features/cluster-classes.md

```diff
@@ -0,0 +1,207 @@
+
+```bash
+> kubectl get kubeadmcontrolplane,machinedeployments
+
+NAME                                                                              CLUSTER                   INITIALIZED   API SERVER AVAILABLE   REPLICAS   READY   UPDATED   UNAVAILABLE   AGE     VERSION
+kubeadmcontrolplane.controlplane.cluster.x-k8s.io/clusterclass-quickstart-XXXX    clusterclass-quickstart   true          true                   1          1       1         0             2m21s   v1.21.2
+
+NAME                                                                             CLUSTER                   REPLICAS   READY   UPDATED   UNAVAILABLE   PHASE     AGE     VERSION
+machinedeployment.cluster.x-k8s.io/clusterclass-quickstart-linux-workers-XXXX    clusterclass-quickstart   1          1       1         0             Running   2m21s   v1.21.2
+
+```
+
+To update the Cluster the only change needed is to the `version` field under `spec.topology` in the Cluster object.
+
+
+Change `1.21.2` to `1.22.0` as below. 
+```bash
+kubectl patch cluster clusterclass-quickstart --type json --patch '[{"op": "replace", "path": "/spec/topology/version", "value": "v1.22.0"}]'
+```
+
+The upgrade will take some time to roll out as it will take place machine by machine with older versions of the machines only being removed after healthy newer versions come online.
+
+To watch the update progress run:
+```bash
+watch kubectl get kubeadmcontrolplane,machinedeployments
+```
+After a few minutes the upgrade will be complete and the output will be similar to:
+```bash
+NAME                                                                              CLUSTER                   INITIALIZED   API SERVER AVAILABLE   REPLICAS   READY   UPDATED   UNAVAILABLE   AGE     VERSION
+kubeadmcontrolplane.controlplane.cluster.x-k8s.io/clusterclass-quickstart-XXXX    clusterclass-quickstart   true          true                   1          1       1         0             7m29s   v1.22.0
+
+NAME                                                                             CLUSTER                   REPLICAS   READY   UPDATED   UNAVAILABLE   PHASE     AGE     VERSION
+machinedeployment.cluster.x-k8s.io/clusterclass-quickstart-linux-workers-XXXX    clusterclass-quickstart   1          1       1         0             Running   7m29s   v1.22.0
+```
+
+## Clean Up
+
+Delete workload cluster.
+```bash
+kubectl delete cluster clusterclass-quickstart
+```
+<aside class="note warning">
+
+IMPORTANT: In order to ensure a proper cleanup of your infrastructure you must always delete the cluster object. Deleting the entire cluster template with `kubectl delete -f clusterclass-quickstart.yaml` might lead to pending resources to be cleaned up manually.
+</aside>
+
+Delete management cluster
+```bash
+kind delete clusters capi-test
+```
+
+<!-- links -->
+[quick start guide]: ../../user/quick-start.md
+[bootstrap cluster]: ../../reference/glossary.md#bootstrap-cluster
+[clusterctl]: ../../clusterctl/overview.md
```

Comment:
This is the clusterctl overview, but the users can't find instruction on how to install it here. should we point to quick-start.html#install-clusterctl instead?